### ISSUE TITLE
Marking owned properties around the map

### ISSUE THREAD

**Issue by raj-patra** [PLTytus/GTAWeb#81]
**Marking owned properties**
...

Currently, when displaying any property, the tool shows all the available property locations for the player to buy but there is no way to mark a particular property as owned.

Suggesting a feature for marking owned properties around the map. This would help the users get an idea about all the properties they own and things happening around it. It could be a quality-of-life update for the tool.
...

**Comment by gaiking-uk** [comment]
### I agree this would be a really useful feature to add (I actually game to GitHub to raise this exact request myself but saw someone else had already requested it).

## Extra context: Goal
It would be great if I can include my own buildings on the GTA Web map, so when I am looking for spray cans or drug dealers, I can easily tell "oh, that's right next to my warehouse in Del Perro" for example...

![Goal -- Map with my buildings](https://github.com/user-attachments/assets/a65c17e5-d8ca-4c0a-bec3-93b45c8024bb)

## Extra context: Current challenge
I have been playing GTA Online for years and so I know the map fairly well, however without as many other buildings shown it can take a little longer to match the shape of the streets or count the number of bridges over the freeway, etc to workout where things are...

![GTA Web -- Current](https://github.com/user-attachments/assets/7ea50afb-1535-4bc3-b886-510a1074eef9)

However if you try to add your own businesses (essentially replicate the in-game map more closely) things go a bit crazy!...

![GTA Web -- With my buildings](https://github.com/user-attachments/assets/82271044-b080-4c83-a572-d0eb24f403c4)

You can fix this _a bit_ by adding things like `LS Meet`, `Luxury Autos`, `Recording Studio`, etc to the map, but this has limited help as there are only a few common locations to add...

![GTA Web -- With public buildings](https://github.com/user-attachments/assets/19d3eb83-8906-426d-a5e9-5dbced0e02ce)

***

## In summary

### If there is an option to add ***MY*** Arcade to the map, and ***MY*** Office to the map -- rather than ***EVERY*** Arcade and ***EVERY*** Office, this would be great and would really appreciate it! 👍🏼👍🏼

**Comment by Gavitron** [comment]
I agree, I just started using the site, and I spent at least 5 minutes trying to figure out how to mark my specific properties, before I realized it was an unimplemented feature.  Maybe something as simple as putting a "mark as owned" button on each info page, and clicking it in one hides the other similar properties when "show only owned" is checked elsewhere.

**Comment by PLTytus** [comment]
Instead of hiding not owned ones and the option "show only owned" I am thinking it would be better to gray out not owned ones (like when your mark collectible as collected) with option "gray out not owned ones" or option "do not gray out owned ones" - depends which behaviour will default one.

In my opinion this will be better solution for properties like apratments or garages, which you can have more than one - by graying out not owned ones, you won't have to uncheck "show only owned option" to mark another apartment as owned.

**Comment by PLTytus** [comment]
I could also implement a limit for properties marked as owned, like you can have only one office, only one vehicle warehoues, etc. So when you will mark Lombank as owned, when you have Arcadius marked as owned - Arcadius will be unmarked. And for cargo warehouses some prompt when you try mark more than 5 as owned.
But I think I would omit this for apartments and garages, we know the limit but there are places like Eclipse Towers, when you can have multiple apartments, or Eclipse Blvd Garage which is not counted for the apartments/garages limit.

**Comment by Gavitron** [comment]
I think you're correct, allowing properties to be greyed out like the collectibles would be a very workable solution.

Would it be easy to implement a workflow where showing a property category at first highlights all properties as it does now, but once you click the equivalent of "cross-off/undo", (ex:  "buy/sell".) on a given property,  then that property would remain highlighted, and the others in the same category would all grey out at once?  Either way, I imagine that you'd only need to add a boolean per property for "isOwned", and could reuse existing categories for the groupings?

Also, re: limits:  Maybe there's no need to immediately implement accurate limits to owned properties, which would allow for edge cases where someone wants to show 2 nightclubs at once, they just mark both as "owned", and the map would simply accomodate their odd use case without issue.  

One other thought that I have about ownership state is the route planner.  You probably already considered this, but I would suggest that the behaviour of the route planner be to ignore unowned properties if the ownership markers are being used.  I know one of my own use cases of the map was to try to show a subset of my owned businesses, and plot the fastest route to hit them all to empty their safes.

**Comment by Gavitron** [comment]
I just re-read your comment about limits, and while I first adressed it from the perspective of not wanting to add more work to the implementation of the ownership feature, I realize I didn't explicitly state that I think your idea is excellent; to have _some_ groups support limits, others not, and the prompt for warehouses too is a great idea.  It would be a great complement to ownership flags, if you don't think it adds too much effort.

**Comment by PLTytus** [comment]
Only thing left to do is impletation of behavior of not-owned markers.
I was thinking about 3 options to choose between:
- show / dim - default behaviour, shows not owned locations, dims them if you have max (for instance when you have 5 warehouses it dims the rest of them)
- hide - hides unowned properties if you own at least one in category (for instance hides all mansion if you own one)
- dim - dim all unowne properties if you own at least one in category

I need think where to put option to toggles between these mode, and how to implement them. The marking as owned system works pretty good, I probably can release it in current stage already (and maybe later I will) - if someone wishes to modify cookies it may break it, but marking a property as owned (or restoring marking in category) should fix the sitution - so I wasn't wasting time to prevent such behavior of too curious users.

**Comment by PLTytus** [comment]
Option to manipulate how not owned properties are displaye is next show/hide for the whole section:
<img width="313" height="198" alt="Image" src="https://github.com/user-attachments/assets/3fec3be0-57b6-4ca5-9d81-09ab9df371e3" /> <img width="165" height="223" alt="Image" src="https://github.com/user-attachments/assets/732c403b-bcb0-4734-a27a-a9140a759395" />
